tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -278,6 +278,99 @@ void toCameraInfo(const TangoCameraIntrinsics& camera_intrinsics,
+void toTango3DR_Pose(const TangoPoseData& tango_pose_data, Tango3DR_Pose* t3dr_pose) {
+  std::copy(std::begin(tango_pose_data.translation),
+            std::end(tango_pose_data.translation),
+            std::begin(t3dr_pose->translation));
+  std::copy(std::begin(tango_pose_data.orientation),
+            std::end(tango_pose_data.orientation),
+            std::begin(t3dr_pose->orientation));
+}
+// Converts Tango3DR_Mesh to visualization_msgs::Marker (TRIANGLE_LIST).
+// See ROS documentation:
+// http://docs.ros.org/api/visualization_msgs/html/msg/Marker.html
+// @param grid_index, index of the tango_mesh.
+// @param tango_mesh, Tango3DR_Mesh to convert.
+// @param mesh_marker, the output visualization_msgs::Marker.
+void toMeshMarker(const Tango3DR_GridIndex& grid_index,
+                  Tango3DR_Mesh* tango_mesh,
+                  double time_offset,
+                  visualization_msgs::Marker* mesh_marker) {
+  mesh_marker->header.frame_id = toFrameId(TANGO_COORDINATE_FRAME_START_OF_SERVICE);
+  mesh_marker->header.stamp.fromSec(tango_mesh->timestamp + time_offset);
+  mesh_marker->ns = "tango";
+  mesh_marker->type = visualization_msgs::Marker::TRIANGLE_LIST;
+  mesh_marker->action = visualization_msgs::Marker::ADD;
+  mesh_marker->pose.orientation.w = 1.0;
+  mesh_marker->scale.x = 1.0;
+  mesh_marker->scale.y = 1.0;
+  mesh_marker->scale.z = 1.0;
+  mesh_marker->color.r = 1.0;
+  mesh_marker->color.g = 1.0;
+  mesh_marker->color.b = 1.0;
+  mesh_marker->color.a = 1.0;
+  mesh_marker->lifetime = ros::Duration(0);
+  // Make unique id from tango mesh indices.
+  mesh_marker->id = grid_index[0];
+  mesh_marker->id *= 37;
+  mesh_marker->id += grid_index[1];
+  mesh_marker->id *= 37;
+  mesh_marker->id += grid_index[2];
+  for (size_t j = 0; j < tango_mesh->num_faces; ++j) {
+    // Add the 3 points of the triangle face.
+    geometry_msgs::Point point;
+    point.x = tango_mesh->vertices[tango_mesh->faces[j][0]][0];
+    point.y = tango_mesh->vertices[tango_mesh->faces[j][0]][1];
+    point.z = tango_mesh->vertices[tango_mesh->faces[j][0]][2];
+    mesh_marker->points.push_back(point);
+    point.x = tango_mesh->vertices[tango_mesh->faces[j][1]][0];
+    point.y = tango_mesh->vertices[tango_mesh->faces[j][1]][1];
+    point.z = tango_mesh->vertices[tango_mesh->faces[j][1]][2];
+    mesh_marker->points.push_back(point);
+    point.x = tango_mesh->vertices[tango_mesh->faces[j][2]][0];
+    point.y = tango_mesh->vertices[tango_mesh->faces[j][2]][1];
+    point.z = tango_mesh->vertices[tango_mesh->faces[j][2]][2];
+    mesh_marker->points.push_back(point);
+    // Add the corresponding colors.
+    std_msgs::ColorRGBA color;
+    color.r = tango_mesh->colors[tango_mesh->faces[j][0]][0] / 255.;
+    color.g = tango_mesh->colors[tango_mesh->faces[j][0]][1] / 255.;
+    color.b = tango_mesh->colors[tango_mesh->faces[j][0]][2] / 255.;
+    color.a = tango_mesh->colors[tango_mesh->faces[j][0]][3] / 255.;
```

Comment:
Add helper conversion functions ArrayToColor and ArrayToPoint. In the future, these could go to a file named 'msg_helpers'.